src/compiler/output-targets/dist-custom-elements/custom-elements-types.ts

```diff
@@ -75,7 +75,16 @@ const generateCustomElementsTypesOutput = async (
             const componentSourceRelPath = relative(config.srcDir, component.sourceFilePath).replace('.tsx', '');
             const componentDTSPath = join(componentsTypeDirectoryRelPath, componentSourceRelPath);
 
-            return `export { ${importName} as ${exportName} } from '${componentDTSPath}';`;
+            const defineFunctionExportName = `defineCustomElement${exportName}`;
+            // Get the path to the sibling typedef file for the current component
+            const localComponentTypeDefFilePath = `./${component.sourceFilePath.split('/').pop()?.replace('.tsx', '')}`;
```

Comment:
Can we tighten this replace up by ensuring we only replace '.tsx' if it occurs at the end of the string? I think that'd look something like:
```suggestion
            const localComponentTypeDefFilePath = `./${component.sourceFilePath.split('/').pop()?.replace(/\.tsx$/, '')}`;
```